src/util/lifecycle_utils.js

```diff
@@ -148,6 +148,11 @@ function build_lifecycle_filter(params) {
      * @param {Object} object_info
      */
     return function(object_info) {
+        // fail if object is a temp file/part or a folder object or a versioned object
```

Comment:
this logic is shared with list-object. and we would need to change in both cases in case there will be a new hidden file/folder added. maybe we should add a utility function to check if file is internal directory?

Reply:
they are both intended to ignore internal directories when going over files. if there is a new internal directory. we would need to ignore it in both

Reply:
indeed, since both places using this function why would we need to change in 2 different places? or I don't get your point...